pkg/intern/intern_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package intern
+
+import (
+	"testing"
+)
+
+func TestIntern(t *testing.T) {
+	p := NewPool(4)
+
+	strings := []string{
+		"a",
+		"b",
+		"c",
+		"d",
+		"e",
+		"f",
+		"g",
+		"h",
+	}
+
+	// We're only testing the semantics here, we're not actually
+	// verifying that interning has taken place. That's too hard to
+	// do in Go...
```

Comment:
I played with that stuff and it unfortunately doesn't work for strings.

The point here would be to know whether a string gets collected or not. Unfortunately, calling SetFinalize on a string gives a runtime error because a string is a value types and not a pointer. I'd really need to be passing the underlying uintptr which I don't have access to.

Anyway, the library code is so simple, I'm pretty sure it's behaving as it should.